### ISSUE TITLE
Lagrange's interpolation for high-order cells

### ISSUE THREAD

**Issue by songhanzhang** [pyvista/pyvista#6968]
### Describe the feature you would like to be added.

I am recently using pyvista for the visualization of my finite element model, and the model consists of 27-node hexahedron elements which are isoparametric. In addition to the vertices, the positions of the points inside the element (including edges and faces) should follow the Lagrange's interpolation.

When I use the feature "pv.CellType.TRIQUADRATIC_HEXAHEDRON" to add the hexahedron cells, however, such interpolation is not considered. The edges are simply connecting the key vertices, which are piecewise linear instead of quadratic interpolation, as shown below:

![Screenshot 2024-12-11 110614](https://github.com/user-attachments/assets/1d72d908-f0ea-47d3-829c-70ffe91e6b2b)

To overcome this issue, I initiated a new feature where the hexahedron is divided into a number of 8-node linear hexahedrons, and the positions of their vertices are evaluated from the quadratic interpolation of the 27-node quadratic hexahedron. As such, all the curved edges and faces can be correctly presented, as shown below.

![Screenshot 2024-12-11 110429](https://github.com/user-attachments/assets/863efaed-d9a8-41ed-8ae8-841f8fb4aa0b)

Can anyone know if such feature has already been included in pyvista? If so, I will prefer to use the existing features avoiding repeating development. If not, I will try to make a pull request for this type of features.

Thank you!

### Pseudocode or Screenshots

```python
Insert code here
```
Or drag your screenshot here!

**Comment by MatthewFlamm** [comment]
Does this example do what you want? https://docs.pyvista.org/examples/01-filter/extract-surface.html

**Comment by songhanzhang** [comment]
> Does this example do what you want? https://docs.pyvista.org/examples/01-filter/extract-surface.html

Yes, that's it! I'll still prefer to use my own version as I need to involve the edges and scalars, and sometimes I need to define the number of intervals from a specified resolution size (not from a specified integer). But anyway, the example you show have already  provided a very good feature for such interpolation. Thank you!

**Comment by MatthewFlamm** [comment]
If you are able to share a 'simple' example of your workflow in a discussion post, then others could benefit.
